### ISSUE TITLE
Consider using AutoValue for all result classes

### ISSUE THREAD

**Issue by serj-lotutovici** [Mauin/RxFingerprint#69]
I would like to be able to log the result objects. Implementing `toString` for each object may become cumbersome to maintain in the future, so AutoValue seams like a good fit here.
For `FingerprintDecyptionResult` [auto-value-redacted](https://github.com/square/auto-value-redacted) can be used to avoid leacage of sensetive data in logs.

**Comment by Mauin** [comment]
Yep, that might make sense. I don't expect it to raise the method count by much so it's something to consider.

**Comment by serj-lotutovici** [comment]
I'll make a PR then later today.
